pylint/checkers/deprecated.py

```diff
@@ -189,6 +204,26 @@ def deprecated_classes(self, module: str) -> Iterable[str]:
         # pylint: disable=unused-argument
         return ()
 
+    def deprecated_attributes(self) -> Container[str]:
+        """Callback returning the deprecated attributes.
+
+        Returns:
+            collections.abc.Container of deprecated attribute names.
+        """
+        return ()
+
+    def check_deprecated_attribute(self, node: astroid.Attribute) -> None:
+        """Checks if the attribute is deprecated."""
+        inferred_expr = safe_infer(node.expr)
+        if not isinstance(inferred_expr, (nodes.ClassDef, nodes.Module)):
+            return
+        attribute_qname = ".".join((inferred_expr.qname(), node.attrname))
+        for deprecated_name in self.deprecated_attributes():
+            if attribute_qname == deprecated_name:
+                self.add_message(
+                    "deprecated-attribute", node=node, args=(node.attrname,)
+                )
```

Comment:
needs `confidence=INFERRED`

Reply:
Can you elaborate I didnt get this ?

Reply:
One of the arguments to add_message is `confidence`, we need to use that argument with the value `interfaces.INFERRED` (which might already be imported) in order to resolve test failures.

Reply:
confidence=interfaces.INFERENCE  i guess you meant inference rather than inferred

Reply:
Yes, thank you!

Reply:
Yes, thank you!